api/v1beta1/hetznerbaremetalhost_types.go

```diff
@@ -342,6 +342,11 @@ func (sts ControllerGeneratedStatus) GetIPAddress() string {
 	return sts.IPv4
 }
 
+// ShouldCapiMachineGetDeleted returns true, if the corresponding capi machine should get deleted.
```

Comment:
I think I would slightly prefer having this defined as method on the scope. WDYT?

Reply:
@janiskemper it gets used in Remediation, too. So we need a place where both parts of the code can access it.